medusa/indexers/indexer_base.py

```diff
@@ -205,6 +209,158 @@ def _set_item(self, sid, seas, ep, attrib, value):  # pylint: disable=too-many-a
             self.shows[sid][seas][ep] = Episode(season=self.shows[sid][seas])
         self.shows[sid][seas][ep][attrib] = value
 
+    def _save_images_by_type(self, image_type, series_id, images):
+        """
+        Save the highest rated images for a show by image type.
+
+        :param image_type: Image type being processed (e.g. `fanart`)
+        :param series: ID of series being processed
+        :param images: Images to be processed
+        """
+        def pop_stats(it, key):
+            """Get the population statistics for a key."""
+            values = list(gen_values_by_key(it, key))
+            num_values = len(values)
+            total = sum(values)
+            mean = total / num_values
+            std_dev = stats.population_standard_deviation(values)
+            return mean, std_dev, values
+
+        def result(item, threshold, mean):
+            """Calculate a score given a threshold and population mean."""
+            if not threshold:
+                threshold = 1  # Prevent division by zero
+            value = item['rating']
+            weight = item['ratingcount']
+            res_index = item['res_index']
+            score_rated = weights.bayesian(weight, value, threshold, mean)
+            weight_score = .5
+            weight_res = .5
+            score_weighted = weight_score * score_rated + weight_res * res_index
+            item['score_rated'] = score_rated
+            item['score_weighted'] = score_weighted
+            return score_weighted, value, weight, item
+
+        def format_result(item):
+            """Format result row for logging output."""
+            row = '{score:>10.3f} {rating:>10.3f} {votes:>6} {res:>15}\t{url}'
+            return row.format(
+                score=item[0],
+                rating=item[1],
+                votes=item[2],
+                res=item[3]['resolution'],
+                url=item[3]['_bannerpath'],
```

Comment:
Same index for `resolution` and `_bannerpath`.

Reply:
yes the fourth index is the image itself.